Integrations/src/main/java/io/deephaven/integrations/python/PythonUtilities.java

```diff
@@ -29,36 +31,81 @@ static PyObject pyListenerFunc(final PyObject pyObject) {
             }
 
             return pyCallable;
+
         } else if (pyObject.hasAttribute("__call__")) {
             return pyObject;
         } else {
             throw new IllegalArgumentException("The Python object specified should either be callable, or a " +
                     "class instance with an onUpdate method");
         }
     }
+     */
+
+    static PyObject pyListenerFunc(final PyObject pyObject) {
+        return pyCallable(pyObject, "onUpdate");
+    }
+
 
     /**
      * Creates a callable PyObject, either using method.apply() or __call__(), if the pyObjectIn has such methods available.
      *
-     * @param pyObjectIn the python object providing the function - must either be callable or have an `apply` attribute
+     * @param pyObject the python object providing the function - must either be callable or have an `apply` attribute
      *                   which is callable.
      * @return pyCallable that can be called directly with arguments using pyCallable.call(...).
      */
-    static PyObject pyCallableGenerator(PyObject pyObjectIn) {
+    /*
+    static PyObject pyApplyFunc(PyObject pyObjectIn) {
         PyObject pyCallable;
         if(pyObjectIn.hasAttribute("apply")){
             pyCallable = pyObjectIn.getAttribute("apply");
+
             if (!pyCallable.hasAttribute("__call__")){
                 throw new IllegalArgumentException("The Python object provided has an apply attribute " +
                         "which is not callable");
             }
-        }else if (pyObjectIn.hasAttribute("__call__")){
-            pyCallable = pyObjectIn;
-        }else{
+
+            return pyCallable;
+
+        } else if (pyObjectIn.hasAttribute("__call__")){
+            return pyObjectIn;
+        } else {
+            throw new IllegalArgumentException("The Python object specified should either be callable, or a " +
+                    "class instance with an apply method");
+        }
+    }
+    */
+
+    static PyObject pyApplyFunc(PyObject pyObject) {
+        return pyCallable(pyObject, "apply");
+    }
+
+    /**
+     * Creates a callable PyObject, either using method.apply() or __call__(), if the pyObjectIn has such methods available.
+     *
+     * @param pyObject      the python object providing the function - must either be callable or have an `apply` attribute
+     *                      which is callable.
+     * @param pyAttribute   the python attribute that provides the callable method.
+     * @return pyCallable that can be called directly with arguments using pyCallable.call(...).
+     */
```

Comment:
javadocs were copied and are inaccurate.  `method.apply()` only works if the second arg is "apply"